src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -471,20 +496,22 @@ public void Test(Operation<T> operation, T objectToApplyTo)
 
                     if (array.Count <= positionInPathAsInteger)
                     {
-                        throw new JsonPatchException<T>(
+                        LogError(
+                            objectToApplyTo,
                             operation,
                             Resources.FormatInvalidIndexForArrayProperty(operation.op, operation.path),
-                            objectToApplyTo);
+                            action);
```

Comment:
Should we stop if we encounter an error or keep going? That's the risk of changing thing to not-throw.